tensorflow_addons/losses/tests/general_loss_test.py

```diff
@@ -0,0 +1,387 @@
+    # differencing. Returning correct gradients is TensorFlow's job, so this is
+    # just an aggressive sanity check in case some implementation detail causes
+    # gradients to incorrectly go to zero due to quantization or stop_gradients
+    # in some op that is used by the loss.
+    for approximate in [False, True]:
+        num_samples = 100000
+        # Normally distributed inputs.
+        x = float_dtype(np.random.normal(size=num_samples))
+        # Uniformly distributed values in (-16, 3), quantized to the nearest
+        # 0.1 and then shifted by 0.05 so that we avoid the special cases at
+        # 0 and 2 where the analytical gradient wont match finite differences.
+        alpha = float_dtype(
+            np.round(np.random.uniform(-16, 3, num_samples) * 10) / 10.0
+        )
+        # Random uniformy distributed values in [0.5, 1.5]
+        scale = float_dtype(np.random.uniform(0.5, 1.5, num_samples))
+        # Compute the loss and its derivative with respect to all three inputs.
+        x, alpha, scale = [tf.convert_to_tensor(z) for z in (x, alpha, scale)]
+        with tf.GradientTape(persistent=True) as tape:
+            for z in (x, alpha, scale):
+                tape.watch(z)
+            loss = general_loss(x, x, alpha, scale, approximate=approximate)
+            d_x, d_alpha, d_scale = [
+                tape.gradient(tf.reduce_sum(loss), z) for z in (x, alpha, scale)
+            ]
+
+        # Assert that the 95th percentile of errors is <= 1e-2.
+        def assert_percentile_close(v1, v2):
+            np.testing.assert_array_less(np.percentile(np.abs(v1 - v2), 95), 1e-2)
+
+        step_size = float_dtype(1e-3)
+        n_x = (
+            general_loss(x + step_size, x + step_size, alpha, scale) - loss
+        ) / step_size
+        n_alpha = (general_loss(x, x, alpha + step_size, scale) - loss) / step_size
+        n_scale = (general_loss(x, x, alpha, scale + step_size) - loss) / step_size
+        assert_percentile_close(n_x, d_x)
+        assert_percentile_close(n_alpha, d_alpha)
+        assert_percentile_close(n_scale, d_scale)
+
+
+testLossfunPreservesDtype(np.float32)
+testDerivativeIsMonotonicWrtX(np.float32)
+testLossIsNearZeroAtOrigin(np.float32)
+testLossIsQuadraticNearOrigin(np.float32)
+testLossIsBoundedWhenAlphaIsNegative(np.float32)
+testDerivativeIsBoundedWhenAlphaIsBelow2(np.float32)
+testDerivativeIsBoundedWhenAlphaIsBelow1(np.float32)
+testAlphaDerivativeIsPositive(np.float32)
+testScaleDerivativeIsNegative(np.float32)
+testLossIsScaleInvariant(np.float32)
+testAlphaEqualsNegativeInfinity(np.float32)
+testAlphaEqualsNegativeTwo(np.float32)
+testAlphaEqualsZero(np.float32)
+testAlphaEqualsOne(np.float32)
+testAlphaEqualsTwo(np.float32)
+testAlphaEqualsFour(np.float32)
+testAlphaEqualsInfinity(np.float32)
+testApproximateLossIsAccurate(np.float32)
+testLossAndGradientsAreFinite(np.float32)
+testGradientMatchesFiniteDifferences(np.float32)
```

Comment:
please use our testing suite described in [CONTRIBUTING.md](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md) to trigger tests in stead of doing this.

If you only want to test against float32, I would suggest remove the parament `float_dtype`. However, since this loss is highly related to the precision issue, I think we can further test against float64. To this end, please utilize the decorator `@pytest.mark.parametrize`. Some useful examples can be found in the following resources.

Moreover, could you use "snake_case" for function name? It'd be better to match other losses' tests.

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/kappa_loss.py
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/tests/kappa_loss_test.py